internal/translator/translate.go

```diff
@@ -26,9 +27,17 @@ import (
 // as it goes.
 // There are currently no fatal conditions (that should not produces a HTTPProxy output)
 // TODO(youngnick) - change this signature to return HTTPProxy, []string, error if we need that.
-func IngressRouteToHTTPProxy(ir *irv1beta1.IngressRoute) (*hpv1.HTTPProxy, []string) {
+func IngressRouteToHTTPProxy(ir *irv1beta1.IngressRoute) (*hpv1.HTTPProxy, []string, error) {
 
 	var warnings []string
```

Comment:
nit: move warnings closer to where it is used, which is going to be line 63 and I think you'll find that warnings is always nil at that point so you can drop it.